lib/server/server.js

```diff
@@ -416,6 +419,73 @@ function configureApp(vetId, platform) {
 
     }).catch(error => {
       console.log('Android apps error', error);
+      reject(error);
+    });
+
+  });
+}
+
+function uploadAndroidSHA256Fingerprint(vetId, fingerprint) {
+  return new Promise((resolve, reject) => {
+    if (!vetId) {
+      reject(new Error('Vet ID is required.'));
+      return;
+    }
+
+    if (!fingerprint) {
+      reject(new Error('SHA Fingerprint is required.'));
+      return;
+    }
+
+    const vet = GlobalVets.findOne({_id: vetId, $or: [{disabled: false}, {disabled: {$exists: false}}]});
+
+    if (!vet) {
+      reject(new Error(`Couldn't find vet for id ${vetId}`));
+      return;
+    }
+
+    if (!Push.FirebaseApps[vet.appIdentifier]) {
+      initializeFirebaseApp(vetId);
+
+      if (!Push.FirebaseApps[vet.appIdentifier]) {
+        reject(new Error(`Firebase sdk for ${vetId} is not initialized.`));
+        return;
+      }
+    }
+
+    const projectManagement = Push.FirebaseApps[vet.appIdentifier].projectManagement();
+
+    if (!projectManagement) {
+      reject(new Error(`Could not initialize FirebaseAdmin.projectManagement for ${vetId}.`));
+      return;
+    }
+
+    listApps(projectManagement, PLATFORMS.ANDROID).then(apps => {
+
+      if (!apps || apps.length === 0) {
+        reject(new Error(`Could't find android app for vet id ${vetId}`));
+      } else if (apps.length === 1) {
+        let shaCert = projectManagement.shaCertificate(fingerprint.replace(/:/g, ''));
+
+        apps[0].addShaCertificate(shaCert)
+          .then(response => {
+            resolve(response);
+          })
+          .catch(error => {
+            if (error && error.errorInfo && error.errorInfo.code === 'project-management/already-exists') {
+              resolve('Requested entity already exists');
```

Comment:
@mjovkovic Can we read existing fingerprint, or check if it exists before trying to add fingerprint?

Reply:
@mslobodan we could, but what would we accomplish with that? We get that info if we try to upload it again :D

Reply:
@mjovkovic OK, I thought if someone already uploaded wrong one, and with next build we want to fix it. But that will never happen as we are setting it through build script. 😀

Reply:
👍